### DISCUSSION TITLE
BUG_ Login via Cloudflare Tunnels 2FA

### DISCUSSION THREAD

**Discussion by FabioEight** [victorbalssa/abacus#90] [Bug]
**Describe the bug**
My Firefly III instance is reachable at firefly.mydomain.com and is protected by Cloudflare ZTNA (https://www.cloudflare.com/products/zero-trust/zero-trust-network-access/).
Basically, Cloudflare adds a 2FA protection (mail TOTP code) on top of the domain. Once you pass the challenge, you are forwarded to the service.
I managed to login via Abacus only by reaching the domain locally. If I try the domain firefly.mydomain.com, I'm asked for the 2FA challenge, I log into Firefly III and I got an error message.

**To Reproduce**
1. Configure Abacus to reach Firefly III public domain
2. Login using the credentials

**Expected behavior**

Remote login.

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: iPhone
 - OS: iOS 16.1

Is it possible to use Abacus via Cloudlfare ZTNA protection? 
Thank you!

**Comment by victorbalssa** [comment]
Hi @FabioEight ,
I'm not familiar with Cloudflare ZTNA, but can you share the error code/message?
basically, the Oauth2 workflow can be broken by `ZTNA` 2FA here, but not sure

**Comment by FabioEight** [comment]
Hello,

Here's the error code:
![image](https://user-images.githubusercontent.com/12196008/203827437-47ac05c9-66bd-4a92-af12-70a17354276a.png)

As far as I know, it should be possible to bypass the Cloudflare ZTNA using a rule intercepting the HTTP header, maybe it could be useful for this case:
https://developers.cloudflare.com/ruleset-engine/rules-language/fields#http-request-header-fields

**Comment by sziyan** [comment]
Im encountering the same error but Im not using ZTNA, just Cloudflare proxy.

I do have Traefik as reverse proxy though.

**Comment by victorbalssa** [comment]
@FabioEight, @sziyan,
I tried with Traefik and everything worked fine.

Could you retry without the Oauth secret and with the last Firefly-iii backend version: `6.0.9`?

**Comment by WojtekK78** [comment]
I'm having exactly the same issues  still existing on `6.0.9`
Setup looks exactly the same so Firefly behind Cloudflare Zero Tunnel where Abacus works perfectly fine in LAN on IP/local domain (Oauth with and without secret), but Abacus is unable to finalize sign in with JSON parse error when trying to sign-in via remote domain firefly.mydomain.com at the same although Firefly `6.0.9`  works ok remotely via Safari and web page using firefly.mydomain.com.
ANy idea how to help to debug this ?

**Reply by victorbalssa** [reply]
@WojtekK78 I'm working on #93 to fix this issue.

**Comment by victorbalssa** [comment]
@FabioEight 
@WojtekK78 
@sziyan 

A new OTA update has been released: 0.6.2 with the Personal Access Token as a authentication option.
Let me know if that solves this issue.
